### PR TITLE
adds my userspace

### PR DESCRIPTION
**Pull request by cfarvidson**
## Description

This PR adds a my userspace and the ability for me to use the userspace on the handwired dactyl manuform.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds my userspace

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl_manuform/5x6/keymaps/cfarvidson/config.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6/keymaps/cfarvidson/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/handwired/dactyl_manuform/5x6/keymaps/cfarvidson/config.h [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+#define MASTER_LEFT
+
+#define XXX KC_NO
+
+#define LAYOUT_cfarvidson_wrapper(\
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6/keymaps/cfarvidson/config.h:7]: This should go into your keymap, as it's only needed there.

File: layouts/community/ergodox/cfarvidson/config.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#pragma once
```
Comment (drashna) [inline layouts/community/ergodox/cfarvidson/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: layouts/community/split_3x6_3/cfarvidson/config.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (drashna) [inline layouts/community/split_3x6_3/cfarvidson/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: users/cfarvidson/cfarvidson.c [hunk 5]
```diff
@@ -0,0 +1,155 @@
+#include "cfarvidson.h"
```
Comment (drashna) [inline users/cfarvidson/cfarvidson.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: users/cfarvidson/cfarvidson.h [hunk 6]
```diff
@@ -0,0 +1,29 @@
+#include QMK_KEYBOARD_H
+#include "keymap_swedish_pro_osx_ansi.h"
+
+#define _BASE 0
+#define _NAV 1
+#define _MOUSE 2
+#define _MEDIA 3
+#define _NUM 4
+#define _SYM 5
+#define _FUN 6
```
Comment (drashna) [inline users/cfarvidson/cfarvidson.h:10]: we heavily recommend using an enum for the layers, as it is easier to maintain long term, and less likely to introduce accidental firmware bloat (empty/unused layers take up space, for instance).

File: users/cfarvidson/config.h [hunk 7]
```diff
@@ -0,0 +1,29 @@
+#pragma once
```
Comment (drashna) [inline users/cfarvidson/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: users/cfarvidson/rules.mk [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+BOOTMAGIC_ENABLE = no     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
```
Comment (fauxpark) [inline users/cfarvidson/rules.mk:11]: Most of these can probably be removed depending on the supported boards.
```suggestion
BOOTMAGIC_ENABLE = no
MOUSEKEY_ENABLE = yes
EXTRAKEY_ENABLE = yes
CONSOLE_ENABLE = no
COMMAND_ENABLE = no
NKRO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
keymap.c files should not be empty.

File: keyboards/handwired/dactyl_manuform/5x6/keymaps/cfarvidson/config.h (hunk 2 above)
Comment (cfarvidson) [reply keyboards/handwired/dactyl_manuform/5x6/keymaps/cfarvidson/config.h:7]: I'm really showing my c skills with this question, but will that not collide if I build another keyboard than the dactyl manuform? (If I add it in `users/cfarvidson/cfarvidson.h` that is)

For the ergodox i created `layouts/community/ergodox/cfarvidson/config.h` and `layouts/community/split_3x6_3/cfarvidson/config.h` with that wrapper layout. But the dactyl manuforms layout was not represented in the `layouts/community`.

Comment (drashna) [reply keyboards/handwired/dactyl_manuform/5x6/keymaps/cfarvidson/config.h:7]: No worries!

The config.h is included in every C file by how qmk has make configured.  So only defines that should be everywhere (such as config settings) should go here. 

Stuff like the keycode define and layout defines shouldn't go here.

Also, technically, the master left isn't needed either, as that is the default behavior.

Comment (cfarvidson) [reply keyboards/handwired/dactyl_manuform/5x6/keymaps/cfarvidson/config.h:7]: I really don't know how to fix this. It the problem just with this file in this file?

I basically did the same thing as 
- https://github.com/cfarvidson/qmk_firmware/blob/master/keyboards/handwired/dactyl_manuform/5x6/keymaps/manna-harbour_miryoku/config.h  
- https://github.com/cfarvidson/qmk_firmware/blob/master/keyboards/handwired/dactyl_manuform/5x6/keymaps/manna-harbour_miryoku/keymap.c

I must be missing something. 😊 

And I'm able to build `ergodox_ez`, `crkbd` and `tbkmini` by using layouts that are available in the `layouts/community`  like `layouts/community/ergodox/cfarvidson/config.h`

Thank you again for looking at this!

**Comment by cfarvidson** [comment]
So I'm closing this MR. I'll keep my userspace in my own fork for now.

@drashna I would really love some tips on how I could keep a layout such as [keyboards/handwired/dactyl_manuform/5x6/keymaps/cfarvidson/config.h](https://github.com/qmk/qmk_firmware/pull/16055/files/0ebea66aec7f1d39325ec9a886aa880ca45c18f9#diff-6df24b537c9b6c622276ea9f77e65602dcacd9fa2681d3e5e4c28f22494cd0b2) in my userspace.

I don't really know how to do it without following the same pattern as:
- https://github.com/cfarvidson/qmk_firmware/blob/master/keyboards/handwired/dactyl_manuform/5x6/keymaps/manna-harbour_miryoku/config.h
- https://github.com/cfarvidson/qmk_firmware/blob/master/keyboards/handwired/dactyl_manuform/5x6/keymaps/manna-harbour_miryoku/keymap.c

**Pull request closed without merging**